kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -43,33 +62,56 @@ static String secretName(KafkaProxy primary) {
-
+        // formatter=off
         return new SecretBuilder()
                 .editOrNewMetadata()
-
-                .withName(secretName(primary))
-                .withNamespace(primary.getMetadata().getNamespace())
+                    .withName(secretName(primary))
+                    .withNamespace(primary.getMetadata().getNamespace())
                 .endMetadata()
-                .withStringData(Map.of(CONFIG_YAML_KEY,
-                        """
-                                adminHttp:
-                                  endpoints:
-                                    prometheus: {}
-                                virtualClusters:
-                                  demo:
-                                    targetCluster:
-                                      bootstrap_servers: TARGET_BOOTSTRAP_SERVERS
-                                    clusterNetworkAddressConfigProvider:
-                                      type: PortPerBrokerClusterNetworkAddressConfigProvider
-                                      config:
-                                        bootstrapAddress: localhost:9292
-                                        brokerAddressPattern: BROKER_ADDRESS_PATTERN
-                                    logNetwork: false
-                                    logFrames: false
-                                """
-                                .replace(
-                                        "TARGET_BOOTSTRAP_SERVERS", primary.getSpec().getBootstrapServers())
-                                .replace("BROKER_ADDRESS_PATTERN", ProxyService.serviceName(primary))))
+                .withStringData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary)))
                 .build();
     }
+
+    String generateProxyConfig(KafkaProxy primary) {
+
+        List<Clusters> clusters = ClustersUtil.distinctClusters(primary);
+        var virtualClusters = clusters.stream()
+                .collect(Collectors.toMap(
+                        Clusters::getName,
+                        cluster -> getVirtualCluster(primary, cluster),
+                        (v1, v2) -> v1, // Dupes handled below
+                        LinkedHashMap::new));
+
+        Configuration configuration = new Configuration(
+                new AdminHttpConfiguration(null, null, new EndpointsConfiguration(new PrometheusMetricsConfig())),
+                virtualClusters,
+                List.of(),
+                List.of(),
+                false);
+
+        ObjectMapper mapper = ConfigParser.createObjectMapper();
+        try {
+            return mapper.writeValueAsString(configuration);
+        }
+        catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static VirtualCluster getVirtualCluster(KafkaProxy primary, Clusters cluster) {
+        var o = ClustersUtil.distinctClusters(primary);
+        var clusterNum = o.indexOf(cluster);
```

Comment:
as you mentioned in the comment its a bit ugly having to recalculate the distinct clusters each time to access the cluster by index, could we instead transform KafkaProxy into something indexable and use that throughout the reconciliation?